src/posix/platform/hdlc_interface.cpp

```diff
@@ -656,6 +667,94 @@ void HdlcInterface::HandleHdlcFrame(otError aError)
     }
 }
 
+otError HdlcInterface::ResetSocket(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    SuccessOrExit(error = WaitForUsbDeviceRenumeration(mRadioUrl.GetPath()));
+
+    CloseFile();
+
+    mSockFd = OpenFile(mRadioUrl);
+    VerifyOrExit(mSockFd != -1, error = OT_ERROR_FAILED);
+
+exit:
+    return error;
+}
+
+otError HdlcInterface::WaitForUsbDeviceRenumeration(const char *aRadioUrlPath)
+{
+    otError error = OT_ERROR_NONE;
+
+#ifdef OPENTHREAD_USE_UDEV
+    struct udev *udev = udev_new();
+    if (!udev)
+    {
+        otLogCritPlat("Can't create udev");
+        return OT_ERROR_FAILED;
+    }
+
+    struct udev_monitor *mon = udev_monitor_new_from_netlink(udev, "udev");
```

Comment:
Again per style guide, 

> - All scoped (i.e. stack) variable declarations should be placed together at the top of the enclosing scope in which they are used.


(some toolchains and analyzers would not be happy if a new variable declaration happens after a jump/goto exit).